src/Akeneo/Pim/Enrichment/Component/Product/Factory/EmptyValuesCleaner/ChainedEmptyValuesCleaner.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Component\Product\Factory\EmptyValuesCleaner;
+
+/**
+ * @author    Anael Chardan <anael.chardan@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class ChainedEmptyValuesCleaner implements ChainedEmptyValuesCleanerInterface
+{
+    /** @var iterable */
+    private $emptyValuesCleaners;
+
+    public function __construct(iterable $emptyValuesCleaners)
+    {
+        $this->emptyValuesCleaners = $emptyValuesCleaners;
+    }
+
+    public function cleanAll(OnGoingCleanedRawValues $onGoingCleanedRawValues): OnGoingCleanedRawValues
+    {
+        /** @var OnGoingCleanedRawValues $result */
+        $result = array_reduce(
+            $this->iterableToArray($this->emptyValuesCleaners),
```

Comment:
This native method do the trick no ?
```suggestion
            iterator_to_array($this->emptyValuesCleaners),
```

Reply:
iterable is different from iterator ;). Sadly php does not have a native method. :/

Reply:
You should a phpdoc comment then in the private method